services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/permissions/PermissionServiceITest.kt

```diff
@@ -50,68 +54,115 @@ class PermissionServiceITest : DatabaseTest() {
 
-        assertEquals(listOf<Int>(), response)
-    }
+            assertEquals(listOf<Int>(), response)
+        }
 
-    @Test
-    fun `getAllowedHankeIds with permissions returns list of IDs`() {
-        val hankkeet = hankeFactory.saveSeveralMinimal(3)
-        hankkeet
-            .map { it.id }
-            .forEach {
-                permissionService.create(
-                    userId = username,
-                    hankeId = it,
-                    kayttooikeustaso = Kayttooikeustaso.KAIKKI_OIKEUDET,
-                )
+        @Test
+        fun `With permissions returns list of IDs`() {
+            val hankkeet = hankeFactory.saveSeveralMinimal(3)
+            hankkeet
+                .map { it.id }
+                .forEach {
+                    permissionService.create(
+                        userId = username,
+                        hankeId = it,
+                        kayttooikeustaso = Kayttooikeustaso.KAIKKI_OIKEUDET,
+                    )
+                }
+
+            val response = permissionService.getAllowedHankeIds(username, PermissionCode.EDIT)
+
+            assertEquals(hankkeet.map { it.id }, response)
+        }
+
+        @Test
+        fun `Returns ids with correct permissions`() {
+            val kaikkiOikeudet = Kayttooikeustaso.KAIKKI_OIKEUDET
+            val hankemuokkaus = Kayttooikeustaso.HANKEMUOKKAUS
+            val hakemusasiointi = Kayttooikeustaso.HAKEMUSASIOINTI
+            val katseluoikeus = Kayttooikeustaso.KATSELUOIKEUS
+            val hankkeet = hankeFactory.saveSeveralMinimal(4)
+            listOf(kaikkiOikeudet, hankemuokkaus, hakemusasiointi, katseluoikeus).zip(hankkeet) {
+                kayttooikeustaso,
+                hanke ->
+                permissionService.create(hanke.id, username, kayttooikeustaso)
             }
 
-        val response = permissionService.getAllowedHankeIds(username, PermissionCode.EDIT)
+            val response = permissionService.getAllowedHankeIds(username, PermissionCode.EDIT)
 
-        assertEquals(hankkeet.map { it.id }, response)
+            assertEquals(listOf(hankkeet[0].id, hankkeet[1].id), response)
+        }
     }
 
-    @Test
-    fun `getAllowedHankeIds return ids with correct permissions`() {
-        val kaikkiOikeudet = Kayttooikeustaso.KAIKKI_OIKEUDET
-        val hankemuokkaus = Kayttooikeustaso.HANKEMUOKKAUS
-        val hakemusasiointi = Kayttooikeustaso.HAKEMUSASIOINTI
-        val katseluoikeus = Kayttooikeustaso.KATSELUOIKEUS
-        val hankkeet = hankeFactory.saveSeveralMinimal(4)
-        listOf(kaikkiOikeudet, hankemuokkaus, hakemusasiointi, katseluoikeus).zip(hankkeet) {
-            kayttooikeustaso,
-            hanke ->
-            permissionService.create(hanke.id, username, kayttooikeustaso)
+    @Nested
+    inner class PermissionsByHanke {
+        private val canView = Kayttooikeustaso.KATSELUOIKEUS
+        private val allRights = Kayttooikeustaso.KAIKKI_OIKEUDET
```

Comment:
I don't really see the need for these aliases?

Reply:
I had them initially to have the asserts on a single line per assert. Nicer for my eyes. But now it looks ok without them. Removed.